### PR TITLE
PLANNER-2525 Change the docs directory structure to comply with Antora

### PR DESCRIPTION
**Pull request by rsynek**
Verified that the generated index.html contains only expected changes. Visually checked the PDF.

### JIRA
https://issues.redhat.com/browse/PLANNER-2525

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

File: optaplanner-docs/src/antora.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+name: optaplanner-docs
+title: OptaPlanner User Guide
+version: latest
```
Comment (rsynek) [inline optaplanner-docs/src/antora.yml:3]: Interestingly enough, the IntelliJ IDEA is capable of rendering images in the docs only if this configuration file is present. It seems the IDEA's asciidoc plugin can detect this file and then assume the typical directory structure (by overriding the `imagesdir` attribute).

Removing unnecessary attributes from .adoc files will be a separate PR.

**Review by triceo — APPROVED** [review]
I did not review every single file, but I did check the resulting PDF and they are identical in the pages where I looked. I followed some links between chapters, they seem to work.

**Comment by rsynek** [comment]
@ge0ffrey I am merging this as we agreed. Should you have some comments, I will address them in a new PR.

**Review by ge0ffrey — COMMENTED** [review]
Great work, thanks Radovan!
